typhos/utils.py

```diff
@@ -493,13 +493,10 @@ def find_templates_for_class(cls, view_type, paths, *, extensions=None,
     elif isinstance(extensions, str):
         extensions = [extensions]
 
-    if len(paths) > 1:
-        paths = remove_duplicate_items(
-            [pathlib.Path(p) for p in paths]
-        )
-
     from .cache import _CachedPath
-    paths = [_CachedPath.from_path(p) for p in paths]
+    paths = remove_duplicate_items(
+        [_CachedPath.from_path(p) for p in paths]
+    )
```

Comment:
Could we move the `remove_duplicate_items` to the `_GlobalDisplayPathCache.add_path`?
So far this function is invoked only at `typhos/display.py:1040` and there `paths` is passed as `paths = cache.get_global_display_path_cache().paths`
That would allow us to do the search for templates without realocation and sorting and cast and only perform the operation when new paths are added. Would that work?

Reply:
It'd work, but it would remove the overall generality of the utility function which I've been trying to retain for future usage 🤷

Reply:
Sounds good! 👍